### DISCUSSION TITLE
An old application isn't working with the new version of MOOSE

### DISCUSSION THREAD

**Discussion by ahmad681** [idaholab/moose#18058] [Q&A General] [answered]
Hi, I am new to the MOOSE environment and a senior graduate student in our group has developed a code in tensor mechanics based off of the older version of MOOSE, 3 years older. With the newer version, there seems to be newer classes that have been deleted or altered and we are afraid of rewriting the code again because there might undermining problems as the MOOSE code is huge. Can you please help me with this issue?

**Comment by loganharbour** [comment, marked as answer]
Can you share the repository with us (if not public, sharing it privately is okay). I can take a quick look to gauge the amount of support needed here.

**Comment by GiudGiud** [comment]
Any news on this?
I would just try compiling it and going through the error messages one by one. There's been a lot of changes, but hopefully a lot of APIs were stabilized already by then.

**Comment by ahmad681** [comment]
Sorry for the late reply Logan and Giud, Yes we figured it out. we went through every single error messege and corrected for it. Thank you
